test/cmdlineTests/message_format/input.sol

```diff
@@ -0,0 +1,103 @@
+// checks that error messages around power-or-10 lines are formatted correctly
+
+
+
+
+
+
+contract C {
+    function f() public { int x; }
+    function g() public { int x; }
+    function h() public {
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
```

Comment:
I think this could be coming down to taste, but instead of having such weird integration tests (like this), I'd suggest to test `SourceReferenceFormatterHuman::printSourceLocation()` by explicitly crafting the `SourceReference` parameter, so a classic unit test. Should be easier to read and only tests want is intended to be tested. What do you think? @chriseth @a3d4?

Reply:
@christianparpart 
> why didn't you just log10(max(_ref.position.line + 1, 1)) + 1 instead?

I did, it was my initial proposal, see 3faaab7e9 :).

There is one issue with `log10` though. Strictly speaking, we cannot guarantee that `log(100)` is `2.0`, and not something like `1.99...9`.

> I think this could be coming down to taste, but instead of having such weird integration tests (like this), I'd suggest to test `SourceReferenceFormatterHuman::printSourceLocation(...)`

I do not have a strong preference here. (Though I agree that the test might look weird. Well, I stopped at 100.) @axic suggested a CLI test, so I implemented it that way.

If people lean towards a classic unit test, I could add it. Or you can provide one, if you feel like doing it.

Reply:
Have a look at [std::round](https://en.cppreference.com/w/cpp/numeric/math/round)